ops/framework.py

```diff
@@ -236,13 +231,18 @@ class EventsBase(Object):
     def __init__(self, parent=None, key=None):
         if parent is not None:
             super().__init__(parent, key)
+        else:
+            self._cache = weakref.WeakKeyDictionary()
 
     def __get__(self, emitter, emitter_type):
         # Same type, different instance, more data. Doing this unusual construct
         # means people can subclass just this one class to have their own 'on'.
         if emitter is None:
             return self
-        return type(self)(emitter)
+        instance = self._cache.get(emitter)
+        if instance is None:
+            instance = self._cache[emitter] = type(self)(emitter)
```

Comment:
The comment at the top of the function is now misplaced. The "unusual construct" is the line right above, not everything else that follows the comment now.

Reply:
Moved.